src/acom_music_box/music_box.py

```diff
@@ -197,20 +197,32 @@ def solve(self, output_path=None, callback=None):
         df = pd.DataFrame(output_array[1:], columns=output_array[0])
         # outputs to file if output is present
         if output_path is not None:
-
-            # Check if the output_path is a full path or just a file name
-            if os.path.dirname(output_path) == '':
-                # If output_path is just a filename, use the current directory
-                output_path = os.path.join(os.getcwd(), output_path)
-            elif not os.path.basename(output_path):
-                raise ValueError(f"Invalid output path: '{output_path}' does not contain a filename.")
-
-            # Ensure the directory exists
-            dir_path = os.path.dirname(output_path)
-            if dir_path and not os.path.exists(dir_path):
-                os.makedirs(dir_path, exist_ok=True)
-
-            df.to_csv(output_path, index=False)
+            if outputformat is None or outputformat == 'csv':
+                # Check if the output_path is a full path or just a file name
+                if os.path.dirname(output_path) == '':
+                    # If output_path is just a filename, use the current directory
+                    output_path = os.path.join(os.getcwd(), output_path)
+                elif not os.path.basename(output_path):
+                    raise ValueError(f"Invalid output path: '{output_path}' does not contain a filename.")
+
+                # Ensure the directory exists
+                dir_path = os.path.dirname(output_path)
+                if dir_path and not os.path.exists(dir_path):
+                    os.makedirs(dir_path, exist_ok=True)
+                
+                #Append csv headers
+                unit_mapping = {
+                    'ENV.temperature': 'K',
+                    'ENV.pressure': 'kPa',
+                    'ENV.number_density_air': 'kg-m3',
+                    'time': 's'
+                }
+                df.columns = [
+                    f"{col} ({unit_mapping[col]})" if col in unit_mapping else 
+                    f"{col} (mol m-3)" if col.startswith('CONC.') else col
```

Comment:
```suggestion
                    f"{col}.({unit_mapping[col]})" if col in unit_mapping else 
                    f"{col}.(mol m-3)" if col.startswith('CONC.') else col
```

We prefer period separated titles